framework/src/source/TestGroup.bs

```diff
@@ -220,6 +228,37 @@ namespace rooibos
       return false
     end function
 
+    private sub notifyReportersOnTestGroupBegin()
```

Comment:
possibly condense based on feedback.

Reply:
Talked to @TwitchBronBron , We will do this later when we consolidating all the code flows as part of supporting promises.